src/garage/torch/algos/sac.py

```diff
@@ -0,0 +1,414 @@
+        # automatic entropy coefficient tuning
+        self.use_automatic_entropy_tuning = use_automatic_entropy_tuning
+        if self.use_automatic_entropy_tuning and not alpha:
+            if target_entropy:
+                self.target_entropy = target_entropy
+            else:
+                self.target_entropy = -np.prod(
+                    self.env_spec.action_space.shape).item()
+            self.log_alpha = torch.Tensor([self._initial_log_entropy
+                                           ]).requires_grad_()
+            self.alpha_optimizer = optimizer([self.log_alpha],
+                                             lr=self.policy_lr)
+        else:
+            self.log_alpha = torch.Tensor([alpha]).log()
+        self.episode_rewards = deque(maxlen=30)
+
+    def train(self, runner):
+        """Obtain samplers and start actual training for each epoch.
+
+        Args:
+            runner (LocalRunner): LocalRunner is passed to give algorithm
+                the access to runner.step_epochs(), which provides services
+                such as snapshotting and sampler control.
+
+        Returns:
+            float: The average return in last epoch cycle.
+
+        """
+        last_return = None
+        for _ in runner.step_epochs():
+            for _ in range(self.steps_per_epoch):
+                if not self._buffer_prefilled:
+                    batch_size = int(self.min_buffer_size)
+                else:
+                    batch_size = None
+                runner.step_path = runner.obtain_samples(
+                    runner.step_itr, batch_size)
+                path_returns = []
+                for path in runner.step_path:
+                    self.replay_buffer.add_transitions(
+                        observation=path['observations'],
+                        action=path['actions'],
+                        reward=path['rewards'],
+                        next_observation=path['next_observations'],
+                        terminal=path['dones'])
+                    path_returns.append(sum(path['rewards']))
+                assert len(path_returns) is len(runner.step_path)
+                self.episode_rewards.append(np.mean(path_returns))
+                for _ in range(self._gradient_steps):
+                    policy_loss, qf1_loss, qf2_loss = self.train_once()
+            last_return = log_performance(runner.step_itr,
+                                          self._obtain_evaluation_samples(
+                                              runner.get_env_copy(),
+                                              num_trajs=10),
+                                          discount=self.discount)
+            self._log_statistics(policy_loss, qf1_loss, qf2_loss)
+            tabular.record('TotalEnvSteps', runner.total_env_steps)
+            runner.step_itr += 1
+
+        return np.mean(last_return)
+
+    def train_once(self, itr=None, paths=None):
```

Comment:
Can you just call this method something else (maybe `_train_once`), and remove the arguments? You're overloading `train()` anyways.